src/Keboola/GoodDataWriter/App.php

```diff
@@ -34,10 +35,36 @@ public function __construct($consoleOutput)
         $this->temp->initRunFolder();
     }
 
+    public function checkProjectAccess(Client $gdClient, ProvisioningClient $provisioningClient, array $config)
+    {
+        try {
+            $gdClient->get("/gdc/projects/{$config['parameters']['project']['pid']}");
+        } catch (Exception $e) {
+            if ($e->getCode() !== 403) {
+                throw $e;
+            }
+
+            if (!getenv('KBC_TOKEN')) {
+                throw new \Exception('KBC Token is missing from the environment');
+            }
+            $provisioningClient->addUserToProject(
+                $config['parameters']['user']['login'],
+                $config['parameters']['project']['pid']
+            );
+        }
+        return true;
+    }
+
     public function run($config, $inputPath)
     {
         Config::check($config);
-        $this->gdClient = $this->initGoodDataClient($config);
+        $gdClient = $this->getGoodDataClient($config);
+        $provisioningClient = new ProvisioningClient(
+            $config['image_parameters']['provisioning_url'],
```

Comment:
`provisioning_url`: Podobny "problem" som riesil aj na ui a tam mi Najlos navrhol ze by sa to mohlo brat z indexu storage(https://connection.keboola.com/v2/storage) zo `services` property ale zatial tam neni. Ta url bude ine v eu regione nie?

Reply:
Je to ve stack parameters, takže EU instance tady dostane EU url. Asi bych to v nich zatím nechal a dal to do issues, to by byla zas tuna práce a předělávání kvůli testům. ;o)